app/models/registration.rb

```diff
@@ -214,12 +218,17 @@ def record_payment(
     receipt,
     user_id
   )
+    # Don't create multiple registration payments for a manual payment
```

Comment:
We could also have logic in `update_status_and_charges` determining when to `yield` - but that logic risks getting clunky (especially when we add in status regressions from uncapturing a payment in MPI 5). This seems like the cleanest way of handling it, although I'm not thrilled. 

The other option would be a `uniqueness` validation, but then to avoid the error being thrown we would need to check for the presence of other manual RegistrationPayments anywhere that `record_payment` is called - so it still seems better to do this check in `record_payment`